src/symbol/placement.js

```diff
@@ -393,6 +394,9 @@ export class Placement {
         }
 
         bucket.sortFeatures(this.transform.angle);
+        if (bucket.featureSortOrder) {
+            this.retainedQueryData[bucket.bucketInstanceId].featureSortOrder = bucket.featureSortOrder.slice();
+        }
```

Comment:
Is the copy of the array necessary?
Would passing `RetainedQueryData` to `bucket.sortFeatures(...)` avoid the need to store the array on the bucket at all?

Reply:
Yeah, having the sort array always live with `RetainedQueryData` is a good idea, and we can avoid the copy.

There is a subtle oddity here -- the sort order will be based on the last time we update opacities, instead of being based on the last time placement happened. It's a little weird that it's not all synced to the same time point, but at least right now, I can't think of why it would be a problem to use a somewhat newer sort order for the same data.

Reply:
Changed.

Reply:
Actually, I realized this change was wrong in some last minute manual testing! Moving to `RetainedQueryData` is actually awkward because `sortFeatures` doesn't necessarily run on every placement (it will early-exit if the bearing hasn't changed). We'd have to make sure to copy `featureSortOrder` from previous placements, and it's actually much simpler to just use the last `featureSortOrder` from the bucket itself. We still get rid of the array copy.